NonUniqueBeanException for java.util.concurrent.ThreadFactory after update to 2.1.0

Hi! My application doesn't start after updating to 2.1.0, cause of the error:

```
13:59:05.511 [main] ERROR io.micronaut.runtime.Micronaut - Error starting Micronaut server: Bean definition [start_error.MyController] could not be loaded: Failed to inject value for parameter [threadFactory] of class: io.micronaut.scheduling.executor.ExecutorFactory

Message: Multiple possible bean candidates found: [java.util.concurrent.ThreadFactory, java.util.concurrent.ThreadFactory]
Path Taken: new MyController([ExecutorService service]) --> new ExecutorFactory(BeanLocator beanLocator,[ThreadFactory threadFactory])
io.micronaut.context.exceptions.BeanInstantiationException: Bean definition [start_error.MyController] could not be loaded: Failed to inject value for parameter [threadFactory] of class: io.micronaut.scheduling.executor.ExecutorFactory

Message: Multiple possible bean candidates found: [java.util.concurrent.ThreadFactory, java.util.concurrent.ThreadFactory]
Path Taken: new MyController([ExecutorService service]) --> new ExecutorFactory(BeanLocator beanLocator,[ThreadFactory threadFactory])
	at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1549)
	at io.micronaut.context.DefaultApplicationContext.initializeContext(DefaultApplicationContext.java:220)
	at io.micronaut.context.DefaultBeanContext.readAllBeanDefinitionClasses(DefaultBeanContext.java:2780)
	at io.micronaut.context.DefaultBeanContext.start(DefaultBeanContext.java:233)
	at io.micronaut.context.DefaultApplicationContext.start(DefaultApplicationContext.java:166)
	at io.micronaut.runtime.Micronaut.start(Micronaut.java:64)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:299)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:285)
	at start_error.Application.main(Application.java:8)
Caused by: io.micronaut.context.exceptions.DependencyInjectionException: Failed to inject value for parameter [threadFactory] of class: io.micronaut.scheduling.executor.ExecutorFactory

Message: Multiple possible bean candidates found: [java.util.concurrent.ThreadFactory, java.util.concurrent.ThreadFactory]
Path Taken: new MyController([ExecutorService service]) --> new ExecutorFactory(BeanLocator beanLocator,[ThreadFactory threadFactory])
	at io.micronaut.context.AbstractBeanDefinition.getBeanForConstructorArgument(AbstractBeanDefinition.java:1035)
	at io.micronaut.scheduling.executor.$ExecutorFactoryDefinition.build(Unknown Source)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1898)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:2647)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:2633)
	at io.micronaut.context.DefaultBeanContext.getBeanForDefinition(DefaultBeanContext.java:2305)
	at io.micronaut.context.DefaultBeanContext.getBeanInternal(DefaultBeanContext.java:2279)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1206)
	at io.micronaut.scheduling.executor.$ExecutorFactory$ExecutorService1Definition.build(Unknown Source)
	at io.micronaut.context.BeanDefinitionDelegate.build(BeanDefinitionDelegate.java:143)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1898)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:2647)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:2633)
	at io.micronaut.context.DefaultBeanContext.getBeanForDefinition(DefaultBeanContext.java:2305)
	at io.micronaut.context.DefaultBeanContext.getBeanInternal(DefaultBeanContext.java:2279)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1245)
	at io.micronaut.context.AbstractBeanDefinition.getBeanForConstructorArgument(AbstractBeanDefinition.java:1013)
	at start_error.$MyControllerDefinition.build(Unknown Source)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1898)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:2647)
	at io.micronaut.context.DefaultBeanContext.loadContextScopeBean(DefaultBeanContext.java:2183)
	at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1543)
	... 8 common frames omitted
Caused by: io.micronaut.context.exceptions.NonUniqueBeanException: Multiple possible bean candidates found: [java.util.concurrent.ThreadFactory, java.util.concurrent.ThreadFactory]
	at io.micronaut.context.DefaultBeanContext.findConcreteCandidate(DefaultBeanContext.java:1987)
	at io.micronaut.context.DefaultApplicationContext.findConcreteCandidate(DefaultApplicationContext.java:434)
	at io.micronaut.context.DefaultBeanContext.lastChanceResolve(DefaultBeanContext.java:2624)
	at io.micronaut.context.DefaultBeanContext.findConcreteCandidateNoCache(DefaultBeanContext.java:2548)
	at io.micronaut.context.DefaultBeanContext.findConcreteCandidate(DefaultBeanContext.java:2478)
	at io.micronaut.context.DefaultBeanContext.getBeanInternal(DefaultBeanContext.java:2259)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1245)
	at io.micronaut.context.AbstractBeanDefinition.getBeanForConstructorArgument(AbstractBeanDefinition.java:1013)
	... 29 common frames omitted

> Task :run FAILED
Caused by: io.micronaut.context.exceptions.DependencyInjectionException: Failed to inject value for parameter [threadFactory] of class: io.micronaut.scheduling.executor.ExecutorFactory


Caused by: io.micronaut.context.exceptions.NonUniqueBeanException: Multiple possible bean candidates found: [java.util.concurrent.ThreadFactory, java.util.concurrent.ThreadFactory]
```

### Steps to Reproduce

Try to inject `@Named(TaskExecutors.SCHEDULED) ExecutorService` into `@Context` scope bean.

### Expected Behaviour

Injection succeeded

### Actual Behaviour

Injection failed

### Environment Information

- **Operating System**: TODO
- **Micronaut Version:** 2.1.0
- **JDK Version:** 11

### Example Application

https://github.com/Im-dex/micronaut-thread-factory-issue

--- a/http-netty/src/main/java/io/micronaut/http/netty/channel/DefaultEventLoopGroupRegistry.java
+++ b/http-netty/src/main/java/io/micronaut/http/netty/channel/DefaultEventLoopGroupRegistry.java
@@ -57,18 +57,6 @@ public class DefaultEventLoopGroupRegistry implements EventLoopGroupRegistry {
     }
 
     /**
-     * Constructs an event loop group thread factory.
-     *
-     * @param configuration The configuration
-     * @return The thread factory
-     */
-    @EachBean(EventLoopGroupConfiguration.class)
-    @BootstrapContextCompatible
-    protected ThreadFactory eventLoopGroupThreadFactory(EventLoopGroupConfiguration configuration) {
-        return new DefaultThreadFactory(configuration.getName() + "-" + DefaultThreadFactory.toPoolName(NioEventLoopGroup.class));
-    }
-
-    /**
      * Constructs an event loop group for each configuration.
      *
      * @param configuration The configuration
@@ -87,7 +75,8 @@ public class DefaultEventLoopGroupRegistry implements EventLoopGroupRegistry {
                                       configuration.getIoRatio().orElse(null)
                               )).orElseThrow(() -> new ConfigurationException("No executor service configured for name: " + executor));
         } else {
-            ThreadFactory threadFactory = beanLocator.getBean(ThreadFactory.class, Qualifiers.byName(configuration.getName()));
+            ThreadFactory threadFactory = beanLocator.findBean(ThreadFactory.class, Qualifiers.byName(configuration.getName()))
+                    .orElseGet(() ->  new DefaultThreadFactory(configuration.getName() + "-" + DefaultThreadFactory.toPoolName(NioEventLoopGroup.class)));
             return eventLoopGroupFactory.createEventLoopGroup(configuration, threadFactory);
         }
     }

--- a/http-server-netty/src/test/groovy/io/micronaut/runtime/ExecutorServiceWithMultipleEventLoopsSpec.groovy
+++ b/http-server-netty/src/test/groovy/io/micronaut/runtime/ExecutorServiceWithMultipleEventLoopsSpec.groovy
@@ -1,0 +1,83 @@
+/*
+ * Copyright 2017-2019 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.runtime
+
+import io.micronaut.context.ApplicationContext
+import io.micronaut.context.annotation.Context
+import io.micronaut.http.annotation.Controller
+import io.micronaut.http.annotation.Get
+import io.micronaut.scheduling.TaskExecutors
+import spock.lang.Specification
+
+import javax.inject.Inject
+import javax.inject.Named
+import java.util.concurrent.ExecutorService
+
+class ExecutorServiceWithMultipleEventLoopsSpec extends Specification {
+
+    void "test ExecutorService service with @Context"() {
+        when:
+            ApplicationContext applicationContext = ApplicationContext.run([
+                    'micronaut.netty.event-loops.default.num-threads': 1,
+                    'micronaut.netty.event-loops.clients.num-threads': 1,
+                    'micronaut.netty.event-loops.abc.num-threads': 1,
+                    'micronaut.netty.event-loops.xyz.num-threads': 1,
+            ])
+
+        then:
+            applicationContext.getBean(MyController1).service != null
+            applicationContext.getBean(MyController2).service != null
+
+        cleanup:
+            applicationContext.close()
+    }
+
+    @Controller
+    static class MyController1 {
+
+        private final ExecutorService service;
+
+        @Inject
+        MyController1(@Named(TaskExecutors.SCHEDULED) ExecutorService service) {
+            this.service = service
+        }
+
+        @Get("/ping")
+        String ping() {
+            return service.toString()
+        }
+
+    }
+
+    @Context
+    @Controller
+    static class MyController2 {
+
+        private final ExecutorService service;
+
+        @Inject
+        MyController2(@Named(TaskExecutors.SCHEDULED) ExecutorService service) {
+            this.service = service
+        }
+
+        @Get("/ping")
+        String ping() {
+            return service.toString()
+        }
+
+    }
+
+}

